src/Raven.Server/Documents/PeriodicBackup/RavenFtpClient.cs

```diff
@@ -65,135 +67,121 @@ public void UploadFile(string folderName, string fileName, Stream stream)
+        internal FtpClient CreateFtpClient(string url, bool keepAlive)
         {
-            var request = (FtpWebRequest)WebRequest.Create(new Uri(url));
-            request.Method = method;
-
-            request.Credentials = new NetworkCredential(_userName, _password);
-            request.EnableSsl = _useSsl;
+            var client = new FtpClient(url);
+            client.Config.ConnectTimeout = -1;
+            client.Credentials = new NetworkCredential(_userName, _password);
             if (_useSsl)
             {
                 var byteArray = Convert.FromBase64String(_certificateAsBase64);
 
                 try
                 {
-                    var x509Certificate = new X509Certificate(byteArray);
-                    request.ClientCertificates = new X509CertificateCollection(new[] { x509Certificate });
+                    var x509Certificate = new X509Certificate2(byteArray);
+                    client.Config.EncryptionMode = FtpEncryptionMode.Explicit;
+                    client.Config.SslProtocols = SslProtocols.Tls12 | SslProtocols.Tls13;
+                    client.Config.ClientCertificates.Add(x509Certificate);
+                    client.ValidateCertificate += (control, e) =>
+                    {
+                        e.Accept = e.PolicyErrors == SslPolicyErrors.None;
+                    };
                 }
                 catch (Exception e)
                 {
                     throw new ArgumentException($"This is not a valid certificate, file name: {_certificateFileName}", e);
                 }
             }
-                
-            request.UsePassive = true;
-            request.UseBinary = true;
-            request.KeepAlive = keepAlive;
 
-            return request;
+            client.Config.SocketKeepAlive = keepAlive;
+            if (_port != null)
+                client.Port = (int)_port;
+            else
+                client.Port = DefaultFtpPort;
+            client.Connect();
+            return client;
         }
 
         public void TestConnection()
         {
             if (_useSsl && string.IsNullOrWhiteSpace(_certificateAsBase64))
                 throw new ArgumentException("Certificate must be provided when using ftp with SSL!");
 
-            ExtractUrlAndDirectories(out var url, out _);
-            var request = CreateFtpWebRequest(url, WebRequestMethods.Ftp.ListDirectory, keepAlive: false);
-
-            try
+            using (var client = CreateFtpClient(_url, keepAlive: false))
             {
-                var response = request.GetResponse();
-                response.Close();
-            }
-            catch (WebException e)
-            {
-                var response = (FtpWebResponse)e.Response;
-                if (response.StatusCode == FtpStatusCode.ActionNotTakenFileUnavailable)
+                try
                 {
-                    // folder doesn't exist
-                    return;
+                    client.Connect();
```

Comment:
maybe auto connect?

Reply:
Done

Reply:
AutoConnect caused problems with the tests were Connect didn't